wordpress-mobile/WordPress-Android
# Issue/13433 add p2 flag to analytics with site

Closes #13433 

iOS PR is [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/15362)

This PR adds a `site_type` property to all the events where we utilize SiteModel to get information about the site (like we do it with `is_jetpack` flag).

This PR does not add a `site_type` property to reader request, since PR with that flag is still in review. I will add it alongside other events.

To test:
- We don't really test Analytic Utils yet, but I added a test for the `AnalyticsSiteType` enum just in case.
- Confirm that the key and parameter names match the ones in the issue.
- Confirm that property added to correct methods.
- Setup a couple of breakpoints and confirm that the property is correctly attached to the tracking event.

cc @aerych 

PR submission checklist:
